packages/babel-plugin-transform-runtime/src/index.js

```diff
@@ -41,6 +42,16 @@ export default declare((api, options, dirname) => {
     absoluteRuntime = false,
   } = options;
 
+  // reason for conditionally including this module:
+  // https://github.com/babel/babel/pull/8616#issuecomment-418154753
+  const includeMathModule =
+    !semver.intersects(`<=7.0.0`, runtimeVersion) &&
+    !semver.intersects(`>=8.0.0`, runtimeVersion);
+
+  if (!includeMathModule) {
+    delete definitions.Math;
```

Comment:
Instead of deleting `Math` when it is not supported, could we only define it when it is supported?. e.g.

```js
// definitions.js

export default runtimeVersion => ({
  builtins: { /* ... */ },
  methods: {
    /* ... */
    Math: semver.satisfies(runtimeVersion, "7.0.1 - 8.0.0") ? {
      acosh: "math/acosh",
      /* ... */
    } : {}
  }
});
```

Reply:
Yeah, we can do that but then `definitions` will be a function now? All its instances inside `index.js` will need to be invoked (line no. 219, 225, etc). Is that okay? Will there be a significant performance boost if we avoid `delete` operation?

Reply:
Okay, I'll make the change

Reply:
One thing to note is, when we make it a function we also need to update https://github.com/babel/babel/blob/bda7cebdb4b5bf0f67022d9739708cb7905e79e1/packages/babel-plugin-transform-runtime/scripts/build-dist.js#L11 and it should just always include all definitions when called in that case.

Reply:
@loganfsmyth @nicolo-ribaudo both the issues are addressed